.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: check-yaml
+    -   id: check-added-large-files
+    -   id: check-executables-have-shebangs
+    -   id: check-merge-conflict
+    -   id: detect-private-key
+    -   id: no-commit-to-branch
```

Comment:
Does this need more arguments to make main protected or is that read from the github config?

Reply:
NO, `main` and `master` branches are protected by default.

Reply:
Great!